l10n_ec_withhold/tests/test_l10n_ec_sale_withhold.py

```diff
@@ -0,0 +1,362 @@
+                self.assertEqual(line.price_unit, line.tax_base_amount)
+
+    def test_l10n_ec_fail_sale_withhold(self):
+        """
+        Fail when document number of withhold not has a valid format
+        or
+        Fail when invoice date not be less of withhold date
+        """
+        partner = self.partner_with_email
+        invoice = self.get_invoice(partner)
+
+        wizard = Form(
+            self.env["l10n_ec.wizard.create.sale.withhold"].with_context(
+                **{"active_ids": invoice.ids}
+            )
+        )
+        self.assertEqual(wizard.partner_id, partner)
+
+        wizard.issue_date = invoice.invoice_date
+        wizard.journal_id = invoice.journal_id
+        wizard.electronic_authorization = "1234567890"
+        # Bad format document number
+        with self.assertRaises(UserError):
+            wizard.document_number = "1"
+
+        tax_group, taxes = self.get_tax_group_and_taxes()
+
+        with wizard.withhold_line_ids.new() as line:
+            line.invoice_id = invoice
+            line.tax_group_withhold_id = tax_group
+            line.tax_withhold_id = taxes
+        # Invoice date not be less of withhold date
+        wizard.issue_date = invoice.invoice_date - relativedelta(days=1)
+
+        with self.assertRaises(UserError):
+            wizard.save().button_validate()
+
+    def test_l10n_ec_authorization_sale_withhold(self):
+        """
+        Fail when authorization key not validate or authorization key not correspond to withhold
+        or
+        Fail when withhold not has detail
+        """
+        partner = self.partner_with_email
+        invoice = self.get_invoice(partner)
+
+        wizard = Form(
+            self.env["l10n_ec.wizard.create.sale.withhold"].with_context(
+                **{"active_ids": invoice.ids}
+            )
+        )
+        self.assertEqual(wizard.partner_id, partner)
+
+        wizard.journal_id = invoice.journal_id
+        with self.assertRaises(UserError):
+            wizard.electronic_authorization = "2809202307123456789000120010020000"
+
+        with self.assertRaises(UserError):
+            wizard.electronic_authorization = (
+                "2809202302123456789000120010020000122171234567815"
+            )
+
+        wizard.electronic_authorization = (
+            "2809202307123456789000120010020000122171234567815"
+        )
+
+        with self.assertRaises(UserError):
+            wizard.save().button_validate()
+
+    def test_l10n_ec_different_invoices_fail_withhold(self):
+        """
+        Fail when select two invoice of different customers
+        """
+        partner1 = self.partner_with_email
+        partner2 = self.partner_ruc
+        invoice1 = self.get_invoice(partner1)
+        invoice2 = self.get_invoice(partner2)
+
+        invoices = self.env["account.move"].search(
+            [
+                ("id", "in", [invoice1.id, invoice2.id]),
+            ]
+        )
```

Comment:
Para simplificar el codigo ya que tienes las facturas como `browse_record` no es necesario hacerle un `search` nuevamente, podrias solo concatenarlas, el ORM admite trabajar con conjunto de registros
```suggestion
        invoices = invoice1 + invoice2
```